### ISSUE TITLE
Editor FPS drops

### ISSUE THREAD

**Issue by adizhavo** [gydisme/Unity-Game-Framwork#4]
Hi, thank you for your code but there seems to be a performance issue with ```HierarchyChangedDetecter``` class.

In project where the hierarchy is quite busy it causes the editor to slow down or even crash.

My solution was to put 
```C#
if(EditorApplication.isPlayingOrWillChangePlaymode) return;
```
when the function ```OnHierarchyChangeCheck``` is called.

Ofc if you want to ignore this you can set up a boolean in the project settings or wrap it in a ```#if bool``` statement 

Hope this helps

**Comment by gydisme** [comment]
thanks!
and I fixed it on https://github.com/gydisme/Unity-HierarchyHelper too! :)

**Comment by gydisme** [comment]
fixed on 209e10ae2eae179b63db0b26da3b77a5340952fd
